AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -46,9 +43,9 @@ final class StorageEngine: StorageEngineBehavior {
         try storageAdapter.setUp(models: StorageEngine.systemModels)
         if #available(iOS 13, *) {
             let syncEngine = isSyncEnabled ? try? RemoteSyncEngine(storageAdapter: storageAdapter) : nil
-            self.init(storageAdapter: storageAdapter, syncEngine: syncEngine, isSyncEnabled: isSyncEnabled)
+            self.init(storageAdapter: storageAdapter, syncEngine: syncEngine)
         } else {
-            self.init(storageAdapter: storageAdapter, syncEngine: nil, isSyncEnabled: isSyncEnabled)
+            self.init(storageAdapter: storageAdapter, syncEngine: nil)
         }
```

Comment:
is there a way to check if Amplify.API is configured here? if it's iOS13 && Amplify.API.isConfigured then create a RemoveSyncEngine with storageAdapter and Amplify.API. I would have imagined RemoteSyncEngine to always need an API when being constructed. 

I don't have a real good use case to expose `isConfigured` at the Category level except that whenever pluginKey's do not match the plugin's key, then the plugin that was added to Amplify doesn't get called on the plugin's configure() method, which developer will then find out it is not configured at plugin's API calls